mambaSharedFramework/HLSValueTypes.swift

```diff
@@ -189,6 +274,109 @@ public struct HLSClosedCaptions: FailableStringLiteralConvertible {
+
+        /// Allows `init` without having to allocate a new `String` object.
+        init?(str: Substring) {
+            switch str {
+            case "BINAURAL": self = .binaural
+            case "IMMERSIVE": self = .immersive
+            case "DOWNMIX": self = .downmix
+            default: return nil
+            }
+        }
+    }
+
+    public init?(string: String) {
+        var count: Int?
+        var spatialAudioCodingIdentifiers: [String]?
+        var specialUsageIdentifiers: [SpecialUsageIdentifier]?
+        let enumeratedSplit = string.split(separator: "/").enumerated()
+        for (index, str) in enumeratedSplit {
+            switch index {
+            case 0: count = Self.parseChannelCount(str: str)
+            case 1: spatialAudioCodingIdentifiers = Self.parseSpatialAudioCodingIdentifiers(str: str)
+            case 2:
+                guard let ids = Self.parseSpecialUsageIdentifiers(str: str) else {
+                    // In the case that we don't recognize one of the special usage identifiers, leading to nil being
+                    // parsed out, I believe it is better to fail the entire parsing, as otherwise we could mislead the
+                    // user of the library into thinking that there are less special usage identifiers than there
+                    // actually are in the CHANNELS attribtue.
+                    return nil
+                }
+                specialUsageIdentifiers = ids
+            default: break // In the future there may be more parameters defined.
+            }
+        }
+        // Count is required to have been parsed.
+        guard let count else {
+            return nil
+        }
+        self.count = count
+        self.spatialAudioCodingIdentifiers = spatialAudioCodingIdentifiers ?? []
+        self.specialUsageIdentifiers = specialUsageIdentifiers ?? []
+    }
+
+    public init(
+        count: Int,
+        spatialAudioCodingIdentifiers: [String],
+        specialUsageIdentifiers: [SpecialUsageIdentifier]
+    ) {
+        self.count = count
+        self.spatialAudioCodingIdentifiers = spatialAudioCodingIdentifiers
+        self.specialUsageIdentifiers = specialUsageIdentifiers
+    }
+
+    private static func parseChannelCount(str: Substring) -> Int? {
+        Int(string: String(str))
+    }
+
+    private static func parseSpatialAudioCodingIdentifiers(str: Substring) -> [String] {
+        let split = str.split(separator: ",")
+        var identifiers = [String]()
+        for id in split where id != "-" {
+            identifiers.append(String(id))
+        }
+        return identifiers
+    }
+
+    private static func parseSpecialUsageIdentifiers(str: Substring) -> [SpecialUsageIdentifier]? {
+        let split = str.split(separator: ",")
+        var identifiers = [SpecialUsageIdentifier]()
+        for id in split {
+            guard let specialUsageId = SpecialUsageIdentifier(str: id) else {
+                return nil
```

Comment:
why return `nil` and not `continue`?

Reply:
My plan is to reject the entire parse if there are any unknown special usage identifiers. This is because I don't want to risk reporting an array with missing values because they couldn't be deserialized into the enum values we know about.

Another option I _could_ take is to add an `unknown(String)` case to the enum and always succeed the parse... The type _could_ be seen as a little more awkward, but may be cleaner... What do you think?

Reply:
I added an `unrecognized(String)` case here: 02d2cb9f8a36379b6b1d49891da3bae364c1309c